pkg/strmr/strmr.go

```diff
@@ -26,23 +26,25 @@ func NewStreamer(samples [][2]float64, numSamples int) *Streamer {
 func (s *Streamer) Stream(samples [][2]float64) (n int, ok bool) {
 	s.mtx.Lock()
 	defer s.mtx.Unlock()
-	
+
 	// Stream is already empty
 	if s.numSamples == 0 {
 		return 0, false
 	}
-	
+
 	numSamplesStreamed := 0
-	for i := range samples {
+
+	maxsamples := len(samples)
+	maxssamplesoffset := len(s.samples) - s.numSamples
```

Comment:
Do I understand it correctly that the purpose of `maxssamplesoffset` is to record where we last left off if we only read a part of the stream in a previous call?